packages/storybook/stories/va-search-input.stories.jsx

```diff
@@ -0,0 +1,126 @@
+import React, { useEffect, useState } from 'react';
+import { VaSearchInput } from '@department-of-veterans-affairs/web-components/react-bindings';
+import { getWebComponentDocs, propStructure } from './wc-helpers';
+import { generateEventsDescription } from './events';
+
+VaSearchInput.displayName = 'VaSearchInput';
+
+const searchDocs = getWebComponentDocs('va-search-input');
+
+export default {
+  title: 'Components/va-search-input',
+  parameters: {
+    componentSubtitle: 'Search Input web component',
+    docs: {
+      description: {
+        component: generateEventsDescription(searchDocs),
+      },
+    },
+  },
+};
+
+const Template = ({ 'button-text': buttonText, value, label, suggestions }) => (
+  <VaSearchInput
+    buttonText={buttonText}
+    value={value}
+    label={label}
+    onInput={e => console.log(e)}
+    onSubmit={e => console.log(e)}
+    suggestions={suggestions}
+  />
+);
+
+const defaultArgs = {
+  'button-text': undefined,
+  'value': 'benefits',
+  'label': 'Search',
+  'suggestions': undefined,
+};
+
+export const Default = Template.bind({});
+Default.args = {
+  ...defaultArgs,
+};
+Default.argTypes = propStructure(searchDocs);
+
+export const WithButtonText = Template.bind({});
+WithButtonText.args = {
+  ...defaultArgs,
+  'button-text': 'Search',
+};
+
+const TypeaheadTemplate = ({ value, suggestions }) => {
+  const [text, setText] = useState(value);
+  const [latestSuggestions, setLatestSuggestions] = useState(suggestions);
+
+  const handleInput = e => {
+    console.log(e);
+    // event.composedPath()[0].value outside of React
+    setText(e.nativeEvent.composedPath()[0].value);
```

Comment:
This feels awkward to me. Could we update `value` on the component as we go so that `e.target.value` will work?

Reply:
See comment in the `value` thread.